rules/react.js

```diff
@@ -1,91 +1,146 @@
-		// Don't allow deprecated React practices
-		"react/no-deprecated": [2, {
-			"react": "0.14.0"
-		}],
+		"react/no-danger": "warn",
+		// Don't allow deprecated React methods.
+		"react/no-deprecated": "error",
 		// Prevent setState in componentDidMount
-		"react/no-did-mount-set-state": [2, "allow-in-func"],
+		"react/no-did-mount-set-state": "error",
 		// Prevent setState in componentDidUpdate
-		"react/no-did-update-set-state": [2, "allow-in-func"],
+		"react/no-did-update-set-state": "error",
 		// Don't allow the state to be mutated. Ever.
-		"react/no-direct-mutation-state": 2,
+		"react/no-direct-mutation-state": "error",
+		// Don't allow findDOMNode, prefer using a callback ref instead.
+		// https://github.com/yannickcr/eslint-plugin-react/issues/678#issue-165177220
+		"react/no-find-dom-node": "error",
 		// Don't allow "isMounted()" calls.
-		"react/no-is-mounted": 2,
+		"react/no-is-mounted": "error",
 		// DISABLING, THERE ARE TIMES WHEN THIS SHOULD BE ALLOWED
 		// Only allow one component to be defined per file.
-		"react/no-multi-comp": 0,
+		"react/no-multi-comp": "off",
+		// Don't use the returned element from rendering with ReactDOM.
+		"react/no-render-return-value": "error",
 		// Allow local state to be changed, even though we're using Redux.
-		"react/no-set-state": 0,
+		"react/no-set-state": "off",
 		// Don't allow string references.
-		"react/no-string-refs": 2,
+		"react/no-string-refs": "error",
+		// Don't allow (usually accidentally) unescaped entities from appearing.
+		"react/no-unescaped-entities": "error",
 		// Don't allow unknown properties.
-		"react/no-unknown-property": 2,
+		"react/no-unknown-property": "error",
+		// Prevent setState in componentDidUpdate
+		"react/no-will-update-set-state": "error",
 		// Only allow ES6 classes.
-		"react/prefer-es6-class": [2, "always"],
+		"react/prefer-es6-class": ["error", "always"],
+		// Prefer stateless functions whenever possible.
+		"react/prefer-stateless-function": "error",
 		// Require prop types to be defined.
-		"react/prop-types": 2,
+		"react/prop-types": "error",
 		// Prevent React not being defined when using JSX
-		"react/react-in-jsx-scope": 2,
-		// Only allows .js extensions
-		"react/require-extension": [2, {
-			"extensions": [".js"]
-		}],
+		"react/react-in-jsx-scope": "error",
+		// Don't require defaultProps for props that are not required.
+		// We often use optional props to selectively hide elements in a component if the prop is not
+		// provided, this would be detrimental to that practice.
+		"react/require-default-props": "off",
+		// Don't require optimization -- yet.
+		// This would require every compnent to have a shouldComponentUpdate defined.
+		"react/require-optimization": "off",
+		// Don't allow people to forget to return inside render.
+		"react/require-render-return": "error",
 		// Don't allow extra closing tags for components without children.
-		"react/self-closing-comp": 2,
+		"react/self-closing-comp": "error",
```

Comment:
I think that's why I feel like it could get confusing - there are some html tags that are self-closing, but there's at least a reasoning about why some are self-closing and others aren't. If we start getting in the habit of doing things like `<div />` my concern is that switching back to normal html we'll get mixed up and start throwing errors. Seems like an unnecessary pollution of the HTML spec.